tests/core/test_graph.py

```diff
@@ -14,27 +14,45 @@
 
 
 def test_init():
-    edge_index = get_random_edge_index(100, 100, 1000)
+    cuda = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
```

Comment:
I don't think this is the way to go for the unit tests. This would mean that on a machine with GPU support only the GPU version will be tested while on a machine without GPU only CPU will be run. Since the automatic GitHub-pipelines do not have access to a GPU, only CPU will be tested there while most of our local machines have a GPU and would not test the CPU version. This can lead to a lot of confusion in my opinion.
Maybe it would be better to have separate GPU tests that are marked as such (similar to the benchmarks https://github.com/pathpy/pathpyG/blob/747c5f97ba755963600ae9d75d4103aaff121f87/tests/benchmark/test_benchmark.py#L27C1-L27C23). We can then exclude them from the automatic pipeline but run them locally as needed.